gammapy/utils/tests/test_random.py

```diff
@@ -2,14 +2,38 @@
 from __future__ import print_function, division
 import numpy as np
 from numpy.testing import assert_allclose
+from astropy.coordinates import Angle
+from astropy.tests.helper import assert_quantity_allclose
 from ..random import sample_sphere, sample_powerlaw, sample_sphere_distance
 
 
 def test_sample_sphere():
+
+    # test general case
     np.random.seed(0)
     lon, lat = sample_sphere(size=2)
-    assert_allclose(lon, [3.44829694, 4.49366732])
-    assert_allclose(lat, [0.20700192, 0.08988736])
+    assert_quantity_allclose(lon, Angle([3.44829694, 4.49366732], 'radian'))
+    assert_quantity_allclose(lat, Angle([0.20700192, 0.08988736], 'radian'))
+
+    # test specify a limited range
+    lon_min = Angle(40., 'degree')
+    lon_max = Angle(45., 'degree')
+    lat_min = Angle(10., 'degree')
+    lat_max = Angle(15., 'degree')
+    lon, lat = sample_sphere(size=10,
+                             lon_range=Angle([lon_min, lon_max]),
+                             lat_range=Angle([lat_min, lat_max]))
+    assert (lon_min <= lon).all() & (lon < lon_max).all()
```

Comment:
Drawback: it doesn't show the values in case of failure. For this reason I also tested `numpy.testing.assert_array_less`, but this has the same problem as #306, an there is no good method in Astropy. I also don't think we should start reimplementing all assertions for quantities, So I'll use the above-mentioned syntax.